src/backend/api/meals.js

```diff
@@ -1,15 +1,150 @@
+// @ts-nocheck
+const { query } = require("express");
 const express = require("express");
+const { max, sum } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+
+
+router.get("/hh", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const rows = await knex('Meal').select('*');
+  
+    response.json(rows);
   } catch (error) {
     throw error;
   }
 });
+// 1.maxPrice	Number	Returns all meals that are cheaper than maxPrice.	api/meals?maxPrice=90
+router.get("/", async (request, response) => {
+  const queryString=request.query;
+    // 1.maxPrice
+    if("maxPrice" in queryString){
+      try{
+      const maxPrice = Number(queryString.maxPrice);
+      if(isNaN(maxPrice)){
+        response.send('maxPrice should be a number ')
+      }else{
+             mealsData = await knex("Meal").where("price", "<", maxPrice);
+         if(MaxPriceMeals.length !== 0){
+          response.send(mealsData);
+         }else {
+          response.status(404).json({Message:'No Meals Found'})
+         }
+       }
+      }catch(error){
+        throw error;
+      }
+       // 2.availableReservations
+    }else if("availableReservations" in queryString){
+      const isTrueSet = (queryString.availableReservations.toLowerCase() === 'true');
+      if(isTrueSet){
+        try {
+         const rows = await knex('Reservation').sum({guests:'Reservation.number_of_guests'})
+        .join('Meal', 'Meal.id' ,'Reservation.meal_id')
+        .select( 'Meal.title','Meal.id','Meal.max_reservations')
+        .groupBy('Reservation.meal_id')
+        .having('Meal.max_reservations','>','guests')
+        if(rows.length !== 0){
+          response.json(rows);
+         }else {
+          response.status(404).json({Message:'No Data Found'})
+         }
+        } catch (error) {
+          throw error;
+        }
+      } // 3.title  like api/meals?title=Indian%20platter
+    }else if("title" in queryString){
+      const isTitle = queryString.title; 
+      if(isTitle){
+        try {
+         const mealsWithTitle = await knex("Meal").select('id','title').where("title", "like", `%${isTitle}%`);
+         if(mealsWithTitle.length !== 0){
+          response.json(mealsWithTitle);
+         }else {
+          response.status(404).json({Message:'No Meals Found'})
+         }
+        } catch (error) {
+          throw error;
+        }
+      }else{
+        response.status(404).json({Message:'Enter some text to search in title'})
+      }
+        // 4.dateAfter like api/meals?dateAfter=2022-10-01
```

Comment:
Just a general comment: the code style looks a bit messy so I would recommend using Prettier or some auto code formatter to clean things up a bit.